--- a/board/hi3518ev200/board.c		2020-08-23 21:16:00.769480300 +0300
+++ b/board/hi3518ev200/board.c		2020-08-23 21:15:53.269593157 +0300
@@ -115,8 +115,19 @@
	return 0;
 }

+void detect_memory(void) {
+       ulong tested_ram = get_ram_size((long *)CFG_DDR_PHYS_OFFSET, CFG_DDR_SIZE)
+               / 1024 / 1024;
+       printf("Detected RAM size: %dMb\n", tested_ram);
+
+       char msize[128];
+       sprintf(msize, "%dM", tested_ram);
+       setenv("totalmem", msize);
+}
+
 int misc_init_r(void)
 {
+       detect_memory();
 #ifdef CONFIG_RANDOM_ETHADDR
	random_init_r();
 #endif
